### PR TITLE
Make percentual app limit configurable via css custom properties

### PR DESCRIPTION
**Pull request by znerol**
Addresses #13079 by introducing css custom properties instead of hard coded values for `usePercentualAppMenuLimit` and `minAppsDesktop`.

Notes on the implemnetation:
* jQuery only handles custom properties in [3.2.0](https://blog.jquery.com/2017/03/16/jquery-3-2-0-is-out/) and later. However, the vanilla approach implemented here is backwards compatible for all supported browsers.
* Admins wishing to override the values may do so by supplying a custom theme `server.css`.

### CODE REVIEW

**Comment by znerol** [comment]
Am I supposed to check-in the built webpack file? I.e. should I add changes inside `core/js/dist/main.js`?

File: core/src/init.js [hunk 1]
```diff
@@ -39,8 +39,9 @@ const resizeMenu = () => {
 	const appList = $('#appmenu li')
 	const rightHeaderWidth = $('.header-right').outerWidth()
 	const headerWidth = $('header').outerWidth()
-	const usePercentualAppMenuLimit = 0.33
-	const minAppsDesktop = 8
+	const appListStyle = window.getComputedStyle(appList.get(0))
+	const usePercentualAppMenuLimit = parseFloat(appListStyle.getPropertyValue('--appmenu-apps-limit-percentual')) || 0.33
```
Comment (skjnldsv) [inline core/src/init.js:43]: Not sure this works on IE :(

Comment (znerol) [reply core/src/init.js:43]: I think [getPropertyValue](https://caniuse.com/#search=getPropertyValue) is fine and browsers without [CSS Variables support](https://caniuse.com/#feat=css-variables) just fall back to the default of `33%`. Is that fallback a problem?

Comment (skjnldsv) [reply core/src/init.js:43]: Nah, it's just IE, don't worry!

**Comment by skjnldsv** [comment]
Jan, you're ok with that?

**Comment by jancborchardt** [comment]
Hi @znerol, thanks a lot for your contribution! :)

This solution is only accessible to expert users, so we should adjust this a bit so it works for everyone.
@skjnldsv and @juliushaertl suggested in the original issue to do it like Chrome with dragging the "More" element to move the limit: https://github.com/nextcloud/server/issues/13079#issuecomment-457567723
→ That looks like the best way forward. Additionally we could have an admin config setting. But custom CSS or theming doesn’t seem to be the right place for this?

**Comment by znerol** [comment]
I actually did read through the whole discussion and some of the linked threads. In order to avoid the start of yet another thread I suggest that we discuss approach / scope in the issue and only tackle technical/implementation details here (or in a next PR). @jancborchardt  do you mind reposting your thoughts about approach in the other issue?

**Review by juliusknorr — APPROVED** [review]
Tested and works like a charm. Really nice improvement. :+1:

**Review by juliusknorr — DISMISSED** [review]
Aw, picked the wrong tab, sorry about that.

**Comment by juliushaertl** [comment]
So I would be kind of ok adding this, but as @jancborchardt mentioned this is not really discoverable so I fear it will end up being untested and break at some point if we change anything in the regard of the menu. We would definitely need a different approach for the chrome-like behavior so that it can be picked up by an app config value.

**Pull request closed without merging**